enhancements/network/dual-stack.md

```diff
@@ -0,0 +1,261 @@
+---
+title: Dual-Stack Networking
+authors:
+  - "@danwinship"
+reviewers:
+  - TBD
+  - "@dcbw"
+  - "@russellb"
+approvers:
+  - TBD
+creation-date: 2020-04-23
+last-updated: 2020-04-23
+status: implementable
+---
+
+# Dual-Stack Networking
+
+This covers adding dual-stack networking support via ovn-kubernetes in OCP.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Customers need IPv6 support. Customers still need IPv4 support too
+though. Dual-stack FTW!
+
+At the moment we expect dual-stack to be more popular with customers
+than single-stack IPv6, even for customers who mostly only want IPv6,
+because in many cases you end up needing 99% IPv6 and 1% IPv4, and
+single-stack IPv6 Kubernetes does not allow that. (You end up needing
+hacks outside of Kubernetes's view, like Multus-based interfaces.)
+
+## Motivation
+
+Customers would like to be able to run workloads with both IPv4 and
+IPv6 connectivity, without sacrificing any of the networking
+functionality provided by Kubernetes. This will allow OpenShift to
+work in a wider variety of network environments and serve a wider
+variety of workloads.
```

Comment:
Also, not a single cloud provider has a v6-capable metadata endpoint, and they "never" will.